packages/react-router/index.js

```diff
@@ -476,7 +476,8 @@ export function useRoutes(routes, basename = '', caseSensitive = false) {
 
   basename = basename ? joinPaths([parentPathname, basename]) : parentPathname;
 
-  let location = useLocation();
+  location = location || useLocation();
```

Comment:
This would violate the hooks-in-conditional rule, since a location might short circuit the `||`. We should move the hook use outside of this to avoid the problem
```suggestion:
  let currentLocation = useLocation();
  location = location || currentLocation;
```

Reply:
I haven't heard about this rule until now. I am fine with this change.

Reply:
It's this one: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

The `||` creates a conditional here, technically.

Reply:
Oh yes, you're right. Haven't thought about this enough.